src/components/org/apache/jmeter/extractor/json/jsonpath/gui/JSONPostProcessorGui.java

```diff
@@ -0,0 +1,190 @@
+            config.setMatchNumbers(matchNumbersField.getText());
+            config.setComputeConcatenation(computeConcatenationField.isSelected());
+        }
+    }
+
+    /**
+     * Implements JMeterGUIComponent.clearGui
+     */
+    @Override
+    public void clearGui() {
+        super.clearGui();
+        refNamesField.setText(""); //$NON-NLS-1$
+        jsonPathExpressionsField.setText(""); //$NON-NLS-1$
+        matchNumbersField.setText(""); //$NON-NLS-1$
+        defaultValuesField.setText(""); //$NON-NLS-1$
+        computeConcatenationField.setSelected(JSONPostProcessor.COMPUTE_CONCATENATION_DEFAULT_VALUE);
+    }
+
+    private void init() {
+        
+        setLayout(new BorderLayout());
+        setBorder(makeBorder());
+
+        Box box = Box.createVerticalBox();
+        box.add(makeTitlePanel());
+        box.add(createScopePanel(true));
+        add(box, BorderLayout.NORTH);
+        add(makeParameterPanel(), BorderLayout.CENTER);
+
+    }
+    
+    private JPanel makeParameterPanel() {
+        refNamesField = new JLabeledTextField(JMeterUtils.getResString("jsonpp_variable_names"));//$NON-NLS-1$
+        jsonPathExpressionsField = new JLabeledTextField(JMeterUtils.getResString("jsonpp_json_path_expressions"));//$NON-NLS-1$
+        matchNumbersField = new JLabeledTextField(JMeterUtils.getResString("jsonpp_match_numbers"));//$NON-NLS-1$
+        defaultValuesField = new JLabeledTextField(JMeterUtils.getResString("jsonpp_default_values"));//$NON-NLS-1$
+        computeConcatenationField = new JCheckBox();//$NON-NLS-1$
+        JPanel panel = new JPanel(new GridBagLayout());
+        GridBagConstraints gbc = new GridBagConstraints();
+        initConstraints(gbc);
+        addField(panel, refNamesField, gbc);
+        resetContraints(gbc);
+        addField(panel, jsonPathExpressionsField, gbc);
+        resetContraints(gbc);
+        addField(panel, matchNumbersField, gbc);
+        resetContraints(gbc);
+        addField(panel, new JLabel(JMeterUtils.getResString("jsonpp_compute_concat")) ,computeConcatenationField, gbc);
+        resetContraints(gbc);
+        gbc.weighty = 1;
+        addField(panel, defaultValuesField, gbc);
+        return panel;
+    }
+    
+    private void addField(JPanel panel, JLabeledTextField field, GridBagConstraints gbc) {
+        List<JComponent> item = field.getComponentList();
+        panel.add(item.get(0), gbc.clone());
+        gbc.gridx++;
+        gbc.weightx = 1;
+        gbc.fill=GridBagConstraints.HORIZONTAL;
+        panel.add(item.get(1), gbc.clone());
+    }
+    
+    private void addField(JPanel panel, JLabel label, JCheckBox checkBox, GridBagConstraints gbc) {
+        panel.add(label, gbc.clone());
+        gbc.gridx++;
+        gbc.weightx = 1;
+        gbc.fill=GridBagConstraints.HORIZONTAL;
+        panel.add(checkBox, gbc.clone());
+    }
+
+    // Next line
+    private void resetContraints(GridBagConstraints gbc) {
```

Comment:
Why is the comment `Next line` when the method name is `resetCon(s)?traints`?
Wouldn't it be simpler to name the method `nextLine` or something similar?